internal/host/host.go

```diff
@@ -1412,3 +1418,99 @@ func (m *Manager) HandleReclaimBootArtifactDownload(ctx context.Context, h *mode
 func (m *Manager) HandleReclaimFailure(ctx context.Context, h *models.Host) error {
 	return m.sm.Run(TransitionTypeReclaimFailed, newStateHost(h), &TransitionArgsUnbindHost{ctx: ctx, db: m.db})
 }
+
+// autorenameHost renames a host if it is necessary and possible. For example, if the name is
+// localhost.localdomain and the MAC address of the main network interface is A8:CD:16:AE:79:01 it
+// will be renamed to a8-cd-16-ae-79-01. Returns a boolean indicating if the host was actually
+// renamed.
+func (m *Manager) autorenameHost(host *models.Host, inventory *models.Inventory) bool {
+	if host.RequestedHostname != "" {
+		return false
+	}
+	if inventory == nil || inventory.Hostname == "" {
+		return false
+	}
+	if !funk.Contains(hostutil.ForbiddenHostnames, inventory.Hostname) {
+		return false
+	}
+	if len(inventory.Interfaces) < 1 {
+		return false
+	}
+	mainNIC, err := m.mainNIC(inventory)
+	if err != nil {
+		m.log.WithError(err).WithFields(logrus.Fields{
+			"host": host.ID,
+			"name": inventory.Hostname,
+		}).Error(
+			"Host hasn't been automatically renamed because an unexpected error " +
+				"ocurred while trying to find the main network interface card",
```

Comment:
As already wrote, as a user I would be quite confused if I've seen `trying to find the main network interface card` error. Maybe something as simple as `error [...] while trying to find a NIC with non-local IP address`? This implicitly carry on the assumption that NIC must have a MAC address should be true for all the types of interfaces we support in our stack.

Reply:
Done, removed the "main NIC" words everywhere.